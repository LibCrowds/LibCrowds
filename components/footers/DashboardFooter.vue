<template>
  <footer id="dashboard-footer" class="footer">
    <div class="container-fluid">
      <p v-html="copyright" class="my-2 mx-1 float-right"></p>
    </div>
  </footer>
</template>

<script>
import siteConfig from '@/siteConfig'

export default {
  data: function () {
    return {
      copyright: `&copy; ${siteConfig.company}, ${new Date().getFullYear()}`
    }
  }
}
</script>

<style lang="scss" scoped>
@import 'src/assets/style/main';

#dashboard-footer {
  color: $gray-600;
  background: $white;
  font-size: $font-size-sm;
  border-top: 1px solid $gray-300;
}
</style>
