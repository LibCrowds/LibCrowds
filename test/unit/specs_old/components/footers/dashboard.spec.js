import BootstrapVue from 'bootstrap-vue'

import { mount, createLocalVue } from '@vue/test-utils'
import DashboardFooter from '@/components/footers/Dashboard'

describe('Dashboard footer', () => {
  let localVue = null
  let wrapper = null

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(BootstrapVue)
    wrapper = mount(DashboardFooter, {
      localVue
    })
  })

  it('renders correctly', () => {
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })
})
