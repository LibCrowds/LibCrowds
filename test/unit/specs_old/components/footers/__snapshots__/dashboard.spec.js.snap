// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard footer renders correctly 1`] = `

<footer id="dashboard-footer"
        data-server-rendered="true"
        class="footer"
>
  <div class="container-fluid">
    <ul class="list-inline my-2 d-flex flex-row align-items-center">
      <li class="list-inline-item mx-1">
        <a href="https://twitter.com/mytwitterhandle"
           class="text-muted"
        >
          <icon name="twitter">
          </icon>
        </a>
      </li>
      <li class="list-inline-item mx-1">
        <a href="mailto:me@example.com"
           class="text-muted"
        >
          <icon name="envelope">
          </icon>
        </a>
      </li>
      <li class="list-inline-item mx-1">
        <a href="https://github.com/Organization/repo"
           class="text-muted"
        >
          <icon name="github">
          </icon>
        </a>
      </li>
    </ul>
    <p class="my-2 mx-1 float-right text-muted">
      &copy; My Company, 2017
    </p>
  </div>
</footer>

`;
